src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,730 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+    // Try to find the intersection at (c) which the Sliproad shortcuts
+    const auto main_road_intersection = getNextIntersection(intersection_node_id, main_road.eid);
 
-    if (next_road_next_intersection.empty())
+    if (!main_road_intersection)
+    {
         return intersection;
+    }
+
+    if (main_road_intersection->intersection.isDeadEnd())
+    {
+        return intersection;
+    }
 
     // If we are at a traffic loop at the end of a road, don't consider it a sliproad
-    if (intersection_node_id == next_intersection_node)
+    if (intersection_node_id == main_road_intersection->node)
+    {
         return intersection;
+    }
 
-    std::unordered_set<NameID> target_road_names;
+    std::vector<NameID> target_road_name_ids;
+    target_road_name_ids.reserve(main_road_intersection->intersection.size());
 
-    for (const auto &road : next_road_next_intersection)
+    for (const auto &road : main_road_intersection->intersection)
     {
         const auto &target_data = node_based_graph.GetEdgeData(road.eid);
-        target_road_names.insert(target_data.name_id);
+        target_road_name_ids.push_back(target_data.name_id);
     }
 
-    for (auto &road : intersection)
+    // TODO: remove geojson debugging
+    NodeID b, c, d;
+    b = intersection_node_id;
+    c = main_road_intersection->node;
+
+    auto sliproad_found = false;
+
+    // Check all roads for Sliproads and assign appropriate TurnType
+    for (std::size_t road_index = 0, last = intersection.size(); road_index < last; ++road_index)
     {
-        if (linkTest(road))
+        if (road_index == *obvious) // obvious main road can't be Sliproad
+            continue;
+
+        auto &sliproad = intersection[road_index]; // this is what we're checking and assigning to
+        const auto &sliproad_data = node_based_graph.GetEdgeData(sliproad.eid);
+
+        // Intersection is orderd: 0 is UTurn, then from sharp right to sharp left.
+        // We already have an obvious index (bc) for going straight-ish.
+        const auto is_right_sliproad_turn = road_index < *obvious;
+        const auto is_left_sliproad_turn = road_index > *obvious;
+
+        // Road at the intersection the main road leads onto which the sliproad arrives onto
+        const auto crossing_road = [&] {
+            if (is_left_sliproad_turn)
+                return main_road_intersection->intersection.getLeftmostRoad();
+
+            if (is_right_sliproad_turn)
+                return main_road_intersection->intersection.getRightmostRoad();
+
+            BOOST_ASSERT_MSG(false, "Sliproad is neither a left nor right of obvious main road");
+            return main_road_intersection->intersection.getLeftmostRoad();
+        }();
+
+        const auto &crossing_road_data = node_based_graph.GetEdgeData(crossing_road.eid);
+
+        // The crossing road at the main road's intersection must not be incoming-only
+        if (crossing_road_data.reversed)
```

Comment:
I am asking myself if we could get away with checking left/right of obvious and not iterating over all roads here.